docs/source/examples/vqe-demo_custom-gate.myst

```diff
@@ -0,0 +1,282 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.10.3
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Variational Quantum Eigensolver improved with Mitiq
+In this example we investigate how Mitiq can help reduce errors
+and improve convergence of a simple VQE problem executed
+on a simulated noisy backend.
```

Comment:
I would add a bit more details here in the initial introduction. For example you can cite ZNE (with a link to the part in the docs where the specific technique is examplained). 

You could also spend a few words, on why VQE was chosen and what it is about, possibly with a reference. 
Maybe we can try to format the reference in the same style of the user guide, so that hopefully it shows up at https://mitiq.readthedocs.io/en/latest/zz_bibliography.html

Reply:
I updated the introduction with a few sentences on ZNE and another small paragraph on VQE.